# Immunoglobulin
  To establish a benchmark dataset, the following steps will be taken. Protein sequences containing the nonstandard amino acid characters"B", "J", "O", "X", "Z" were first removed. Second to avoid overfitting caused by homology bias and to reduce redundancy, the CD-Hit program was chosen to set a 60% sequence identity cutoff to remove highly similar sequences. Finally, if a certain protein sequence was a subsequence of other proteins it was also removed. Considering that to avoid the influence of the expression of different protein sequences on the predicted effects, we selected only human, mouse, and rat samples.
   For the benchmark dataset,  the sequence information of 109 immunoglobulins was stored in the ip.fasta file. Meanwhile, the sequence information of 119 non-immunoglobulins is stored in the in.fasta file.
   To take  the same steps to download 40 protein sequences that differed from the training set from the Uniprot database as an independent test set data, which included 20 immunoglobulin sequences and 20 non-immunoglobulin sequences. They were stored in iip.fasta and iin.fasta, respectively.
